src/dolphin/iniFile.ts

```diff
@@ -0,0 +1,278 @@
+/**
+ * rewrite of Inifile.cpp/Inifile.h from dolphin in Typescript
+ * https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Common/IniFile.cpp
+ */
+
+import fs from "fs";
+import readline from "readline";
+
+/**
+ * The IniFile Class, contains a Section subclass
+ */
+export class IniFile {
+  private sections: Section[];
+
+  public constructor() {
+    this.sections = [];
+  }
+
+  /** Differs from IniFile.cpp via:
+   * Instead of editing keyOut and valueOut by reference, return them */
+  public parseLine(line: string): Array<string | null> {
```

Comment:
```suggestion
  public parseLine(line: string): [string, string] | [null, null] {
```
I don't think `Array<string | null>` is the correct type to return here.